File: 2206 divide-array-into-equal-pairs/solution.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        vector<int> f(600,0);
        for(int i:nums) f[i]++;
        for(int i:nums) {
            if((f[i] & 1) == 1) return false;
        }
        return true; 
    }
};

File: 2206 divide-array-into-equal-pairs/solution.rs
impl Solution {
    pub fn divide_array(nums: Vec<i32>) -> bool {
        let mut f = vec![0;600];
        for &i in &nums {
            f[i as usize]+=1;
        }
        for &i in &nums {
            if (f[i as usize]) & 1 == 1 {
                return false
            }
        }
        true
    }
}
